sceptre/stack.py

```diff
@@ -281,6 +281,17 @@ def _ensure_parameters(
     ) -> Dict[str, Union[str, List[Union[str, Resolver]], Resolver]]:
         """Ensure CloudFormation parameters are of valid types"""
 
+        def cast_value(value: Any) -> Union[str, List[Union[str, Resolver]], Resolver]:
```

Comment:
This function's purpose was only to validate parameters, as indicated in this methods docstring.  Could we update the docstring to reflect that this method is also casting parameters?  Or what might be even more clear is to split validation and casting actions into two separate methods?

Reply:
@zaro0508 I see your point. On looking at this, splitting it up into 2 or 3 methods feels messier to me than sticking with inner functions but fixing the naming. To avoid the inconsistency between `_ensure_boolean` and `_ensure_parameters`, which now would do different things, I've renamed as `_cast_parameters` and fixed the docstring. What do think about this version?